blocks/subscriptions-block/components/useOffer.jsx

```diff
@@ -0,0 +1,61 @@
+import { useState, useEffect, useCallback } from "react";
+import { useFusionContext } from "fusion:context";
+import getProperties from "fusion:properties";
+import { isServerSide } from "@wpmedia/arc-themes-components";
+
+const offerService = ({ origin, campaignCode = "default", endpoint }) =>
+	fetch(`${origin}${endpoint}${campaignCode || "default"}`, {}).then((res) => res.json());
+
+const useOffer = ({ campaignCode }) => {
+	const { arcSite } = useFusionContext();
+	const {
+		api: {
+			retail: { origin, endpoint },
+		},
+	} = getProperties(arcSite);
+	const [offer, setOffer] = useState(null);
+	const [error, setError] = useState(null);
+	const [isFetching, setIsFetching] = useState(true);
+
+	const fetchOffer = useCallback(
+		async (code) => {
+			try {
+				const offerResponse = await offerService({
+					campaignCode: code,
+					origin,
+					endpoint,
+				});
+				setOffer(offerResponse);
+				return offerResponse;
+			} catch (err) {
+				setError(`Error in fetching retail offers: ${err.toString()}`);
+			}
+			return null;
+		},
+		[origin, endpoint]
+	);
+
+	useEffect(() => {
+		const fetchNewOffer = async () => {
+			setIsFetching(true);
+			await fetchOffer(campaignCode);
+			setIsFetching(false);
+		};
+		if (!offer && !isServerSide()) {
+			if (!origin || !endpoint) {
+				setError("Origin or endpoint properties not set in api.retail for this environment.");
+			} else {
+				fetchNewOffer();
+			}
+		}
+	}, [campaignCode, endpoint, fetchOffer, offer, origin]);
+
+	return {
+		error,
```

Comment:
Could this error actually be output somewhere? The only code I see using it is the test component in the unit test file. Right now this just fails silently.

Reply:
Our team have plans to do a second phase for all this work, including some re-design, since we want to include some payment methods ([figma file](https://www.figma.com/file/wQd5edf6Nt1MB3b6Qlzp6x/Arc-Blocks-Design-System?type=design&node-id=814-31316&mode=design&t=farOToMuRrzuhDwi-0)).
For now, the scope of our tickets is to move subs components from v1 to v2, keeping the same style and functionality. I'll check this concern in our second phase.